File: backend/src/main/java/fr/pilagors/sc/entity/Recipe.java
package fr.pilagors.sc.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import java.util.Set;

@Entity
@Table(name = "recipes")
@Data
@EqualsAndHashCode(exclude = {"ingredients", "products"})
public class Recipe {
    @Id
    private String id;
    private String name;
    private Double duration;
    @Column(name = "is_alternate")
    private Boolean isAlternate;

    @OneToMany(mappedBy = "recipe", fetch = FetchType.LAZY)
    private Set<RecipeIngredient> ingredients;

    @OneToMany(mappedBy = "recipe", fetch = FetchType.LAZY)
    private Set<RecipeProduct> products;
}

File: frontend/src/types/recipe.ts
import type { Item } from './item'

export interface RecipeIngredient {
  id: number
  item: Item
  amount: number
}

export interface Recipe {
  id: string
  name: string
  duration: number
  isAlternate: boolean
  ingredients: RecipeIngredient[]
  products: RecipeIngredient[]
}
